src/lib/swapper/swappers/CowSwapper/endpoints.ts

```diff
@@ -0,0 +1,183 @@
+import { fromAssetId, fromChainId } from '@shapeshiftoss/caip'
+import type { EvmChainId } from '@shapeshiftoss/chain-adapters'
+import { toAddressNList } from '@shapeshiftoss/chain-adapters'
+import type { ETHSignMessage } from '@shapeshiftoss/hdwallet-core'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { Result } from '@sniptt/monads/build'
+import { getConfig } from 'config'
+import { ethers } from 'ethers'
+import { v4 as uuid } from 'uuid'
+import type {
+  GetEvmTradeQuoteInput,
+  GetTradeQuoteInput,
+  GetUnsignedTxArgs,
+  SwapErrorRight,
+  Swapper2Api,
+  TradeQuote2,
+} from 'lib/swapper/api'
+import { assertGetEvmChainAdapter, createDefaultStatusResponse } from 'lib/utils/evm'
+
+import { isNativeEvmAsset } from '../utils/helpers/helpers'
+import { getCowSwapTradeQuote } from './getCowSwapTradeQuote/getCowSwapTradeQuote'
+import type {
+  CowSignTx,
+  CowSwapGetTradesResponse,
+  CowSwapGetTransactionsResponse,
+  CowSwapQuoteResponse,
+} from './types'
+import {
+  COW_SWAP_NATIVE_ASSET_MARKER_ADDRESS,
+  COW_SWAP_SETTLEMENT_ADDRESS,
+  DEFAULT_APP_DATA,
+  ERC20_TOKEN_BALANCE,
+  ORDER_KIND_SELL,
+} from './utils/constants'
+import { cowService } from './utils/cowService'
+import type { CowSwapOrder } from './utils/helpers/helpers'
+import {
+  domain,
+  getCowswapNetwork,
+  getNowPlusThirtyMinutesTimestamp,
+  hashOrder,
+} from './utils/helpers/helpers'
+
+const tradeQuoteMetadata: Map<string, { chainId: EvmChainId }> = new Map()
+
+export const cowApi: Swapper2Api = {
+  getTradeQuote: async (
+    input: GetTradeQuoteInput,
+    { sellAssetUsdRate, buyAssetUsdRate }: { sellAssetUsdRate: string; buyAssetUsdRate: string },
+  ): Promise<Result<TradeQuote2, SwapErrorRight>> => {
+    const tradeQuoteResult = await getCowSwapTradeQuote(input as GetEvmTradeQuoteInput, {
+      sellAssetUsdRate,
+      buyAssetUsdRate,
+    })
+
+    return tradeQuoteResult.map(tradeQuote => {
+      const { receiveAddress, affiliateBps } = input
+      const id = uuid()
+      tradeQuoteMetadata.set(id, { chainId: tradeQuote.steps[0].sellAsset.chainId as EvmChainId })
+      return { id, receiveAddress, affiliateBps, ...tradeQuote }
+    })
+  },
+
+  getUnsignedTx: async ({ from, tradeQuote, stepIndex }: GetUnsignedTxArgs): Promise<CowSignTx> => {
+    const { accountNumber, buyAsset, sellAsset, sellAmountBeforeFeesCryptoBaseUnit } =
+      tradeQuote.steps[stepIndex]
+    const { receiveAddress } = tradeQuote
+
+    const buyToken = !isNativeEvmAsset(buyAsset.assetId)
```

Comment:
This is copied and pasted from the previous implementation but can we use the `address` vernacular here? 🙏🏽